components/unread-articles-indicator/device-session.js

```diff
@@ -0,0 +1,23 @@
+import { isBefore, addMinutes } from 'date-fns';
+import * as storage from './storage';
+const SESSION_THRESHOLD_MINUTES = 30;
+
+class DeviceSession {
+
+	constructor () {
+		this.expiry = storage.getDeviceSessionExpiry();
+		const newExpiry = addMinutes(new Date(), SESSION_THRESHOLD_MINUTES).toISOString();
+		storage.setDeviceSessionExpiry(newExpiry);
+	}
+
+	isWithinSession (targetTimestamp) {
+		if (this.expiry) {
+			return isBefore(targetTimestamp, this.expiry) ? true : false;
```

Comment:
The ternary is redundant - this does the same thing:

```suggestion
			return isBefore(targetTimestamp, this.expiry);
```

(`isBefore` returns a boolean anyway)